tests/main/userns/task.yaml

```diff
@@ -0,0 +1,120 @@
+summary: userns restrictions work for strict mode snaps
+
+details: |
+    This test checks that a snap which plugs the userns interface is able to use
+    unprivileged user namespaces only when the userns interface is connected on
+    systems that support restrictions on unprivileged user namespaces.
+
+# This test is not executed on a core system simply because of the hassle of
+# building the support C program.
+# ubuntu-14.04: the test sets up a user session, which requires more recent systemd
+systems: [-ubuntu-core-*, -ubuntu-14.04-*]
+
+environment:
+    # This is used to abbreviate some of the paths below.
+    P: /var/snap/test-snapd-userns/common
+
+prepare: |
+    echo "Install a helper snap (for confinement testing)"
+    "$TESTSTOOLS"/snaps-state install-local test-snapd-userns
+
+    echo "Compile and prepare the support program"
+    # Because we use the snap data directory we don't need to clean it up
+    # manually as all snaps and their data are reset after each test.
+    # Build the test binary statically, as it will be running inside a base with
+    # potentially older glibc.
+    gcc -Wall -Wextra -Werror ./unshare.c -o "$P/unshare" -static
+
+    tests.session -u test prepare
+
+restore: |
+    tests.session -u test restore
+
+execute: |
+    # ensure that the system supports unprivileged userns and it is enabled
+    if ! sysctl kernel.unprivileged_userns_clone; then
+      echo "Skipping testing of unprivileged userns since it is not supported by the kernel"
+      exit 0
+    fi
+
+    echo "Enable the use of unprivileged user namespaces by default"
+    sysctl -w kernel.unprivileged_userns_clone=1
+
+    echo "The userns interface is disconnected by default"
+    snap connections test-snapd-userns | MATCH 'userns +test-snapd-userns:userns +- +-'
+
+    # running unshare -U to create a new user namespace should be denied as an
+    # unprivileged user by seccomp
+    echo "When the userns interface is disconnected creating a new user namespace is denied"
+    tests.session -u test exec sh -c "snap run test-snapd-userns.with-userns-plug -c '\\$P/unshare -U id'" 2>&1 | MATCH 'unshare: unshare: Operation not permitted'
+
+    echo "Connect the userns interface"
+    snap connect test-snapd-userns:userns
+    snap connections test-snapd-userns | MATCH 'userns +test-snapd-userns:userns +:userns +manual'
+
+    # running unshare -U to create a new user namespace should be allowed
+    echo "When the userns interface is connected creating a new user namespace is allowed"
+    # on arch the group is called nobody
+    if os.query is-arch-linux; then
+      tests.session -u test exec sh -c "snap run test-snapd-userns.with-userns-plug -c '\\$P/unshare -U id'" 2>&1 | MATCH 'uid=65534\(nobody\) gid=65534\(nobody\) groups=65534\(nobody\)'
```

Comment:
perhaps it is simpler to create a var called group=nobody and update in case or arch linux and then just run tests.session once in the code

Reply:
Good idea - done in https://github.com/snapcore/snapd/pull/12844/commits/daf01f16fb91d875311c48df0d190b3853b253bc